### ISSUE TITLE
`petite-vue-i18n`: `flatJson` bug (update: actually confusion, not bug)

### ISSUE THREAD

**Issue by NamesMT** [intlify/vue-i18n#2011]
### Reporting a bug?

~~`petite-vue-i18n` cannot work with `flatJson` enabled, but when disabled, it works fine.~~
See #2010

TL;DR: `petite-vue-i18n` by default only works with a `key:value` (flat) json messages format, `flatJson` will recursively transforms a flat json key into a nested key, which will broke `petite-vue-i18n` when resolving messages, this is a problem of `flatJson` description rather than `petite-vue-i18n`'s bug.

### Expected behavior

.

### Reproduction

```ts
import { createI18n } from 'petite-vue-i18n'

// Initialize i18n  
const i18n = createI18n({
  legacy: false,
  flatJson: true,
  locale: 'en',
  fallbackLocale: 'en',
  fallbackWarn: false,
  missingWarn: false,
  messages: {
    en: { 'word.hello': 'Hello' },
  },
})

const test = i18n.global.t('word.hello')

console.log({ test })
```

Swapping to `vue-i18n` works fine.

### System Info

```shell
.
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussions](https://github.com/intlify/vue-i18n/discussions)

**Comment by kazupon** [comment]
Thank you for the report and for updating the title to clarify!
Sorry my lately reply for you.

As you discovered, this is expected behavior, not a bug.

- **`petite-vue-i18n`** uses a simple key-value resolver (`resolveWithKeyValue`) by default, which does flat `message[key]` lookups. So `{ 'word.hello': 'Hello' }` works out of the box without `flatJson`.
- **`flatJson: true`** transforms flat keys like `'word.hello'` into nested structures `{ word: { hello: 'Hello' } }` at setup time. This is designed for the full `vue-i18n` build which has a nested path resolver (`resolveValue`) that can traverse `word.hello` as `obj.word.hello`.

Since `petite-vue-i18n` doesn't include the nested path resolver, the transformed nested structure cannot be resolved — making `flatJson: true` incompatible with `petite-vue-i18n`.

We'll improve the documentation to clarify that `flatJson` should not be used with `petite-vue-i18n`. Closing this issue.

close, Thanks!
